src/main/java/dk/kb/datahandler/kaltura/KalturaDeltaUploadJob.java

```diff
@@ -0,0 +1,304 @@
+                 
+                   String fileError= hasStreamFileError(path, minimumFileSizeInBytes);
+                   if (fileError != null) {
+                       log.warn("File does not exist='{}' or size in bytes less than '{}'. Error='{}'. Id='{}'. Skipping upload", path, minimumFileSizeInBytes,fileError,id);     
+                       updateKalturaIdForRecord(storageClient, fileId, fileError);
+                       continue;                       
+                   }
+                   
+                   // Check file not already in kaltura. 
+                   String kalturaInternalId=getInternalIdKaltura(fileId);
+                   if (kalturaInternalId != null) {
+                       log.warn("Stream allready found in kaltura. FileId='{}'. Setting this kalturaId for record.",kalturaInternalId  );
+                       updateKalturaIdForRecord(storageClient, fileId, kalturaInternalId);                       
+                       continue; //Do not upload stream
+                   }
+                   
+                   try {
+                     String kalturaId=uploadStream(title, fileId, description, path, uploadTagForKaltura,mediaType,flavourParamId);
+                     log.info("Upload stream='{}' and got kalturaId='{}'",path,kalturaId);
+                     numberStreamsUploaded++; //Success count
+                     //update storage record with kalturaId                     
+                     updateKalturaIdForRecord(storageClient, fileId, kalturaId);
+                     log.info("Updated kaltura mapping in storage for fileId:"+fileId);  
+                     continue; //Not required since nothing else happens below. But for consistency.
+                   }
+                   catch(Exception e) {  //Stop delta job
+                       log.error("Failed uploading stream to kaltura with fileId='{}', path='{}', title='{}', error='{}'",fileId,path,title,e.getMessage());
+                       updateKalturaIdForRecord(storageClient, fileId, StreamErrorTypeDto.API.getValue()); //Mark as API error
+                       throw new InternalServiceException("Failed uploading stream to kaltura with fileId="+fileId);
+                   }                                    
+                 }
+                 catch(Exception e) { 
+                   //Totally stop all uploads if a single call fails. Change strategy if this does seem to happen sporadic
+                   //Delta upload can be started again. We want to detect this error and not ignore it.
+                   log.error("Error kaltura lookup for fileId:"+fileId,e);
+                   //Do not mark record with error. We need to know why this happens.
+                   throw new InternalServiceException("Error kaltura lookup for fileId:"+fileId);                         
+                 }                                      
+                }
+                // Check file not already in kaltura
```

Comment:
This comment seem to be hanging without code, my guess is that it can be deleted as it's duplicated on line 121

Reply:
deleted